tests/download/test_session.py

```diff
@@ -34,3 +37,46 @@ def test_token_info(session):
 
     for key in ["sub", "aud", "jti", "exp", "name", "email", "sid", "org", "did", "aid", "d"]:
         assert key in info
+
+
+def test_session_with_missing_credentials(fake_token):
+    config_without_credentials = SHConfig()
+    config_without_credentials.reset()
+
+    with pytest.raises(ValueError):
+        SentinelHubSession(config=config_without_credentials)
+
+    with pytest.raises(ValueError):
+        SentinelHubSession(config=config_without_credentials, _token=fake_token)
+
+    session = SentinelHubSession(config=config_without_credentials, refresh_before_expiry=None, _token=fake_token)
```

Comment:
why does the L50 one fail but L52 one work?

Reply:
Added a comment about that.